### PR TITLE
text validity updates

### PR DESCRIPTION
**Pull request by hwigmore**
[Clubhouse](https://app.clubhouse.io/floored/story/11286/textarea-give-validity-check-power-back-to-the-owner-caller-of-the-component)

context for this: previously i made made this change (#63) so that the owner of the textfield component can choose when to call its validity function, and just let the blocks component know the result (maybe the owner wants to call it on keydown and blur, but maybe they don't, and maybe they wanna call it on a form submit or something).

this pr is to make that same update on `TextArea`

### CODE REVIEW

File: react/input/TextArea.jsx [hunk 1]
```diff
@@ -6,23 +6,11 @@ class TextArea extends React.Component {
...
 
-    this.state = {
-      isValid: props.isValid(props.value)
-    };
-
     this.textAreaRef = React.createRef();
   }
 
   componentDidMount() {
-    if (this.props.isMultiline) {
```
Comment (hwigmore) [inline react/input/TextArea.jsx]: unrelated: removed the `isMultiline` prop, which i think was a relic of of textfield and textarea sharing a component. resize now gets called on mount

react component preview on master
screen shot 2018-12-07 at 11 57 06 am



on branch
screen shot 2018-12-07 at 11 56 48 am

File: react/input/TextArea.jsx [hunk 2]
```diff
@@ -34,46 +22,14 @@ class TextArea extends React.Component {
     this.textAreaRef.current.style.height = `${this.textAreaRef.current.scrollHeight}px`;
   }
 
-  getTextArea() {
-    return (
-      <textarea
```
Comment (hwigmore) [inline react/input/TextArea.jsx]: just moved this into the `render` function to simplify

File: react/input/TextFieldWithDropdown.jsx [hunk 3]
```diff
@@ -22,7 +22,7 @@ TextFieldWithDropdown.propTypes = {
     placeholder: PropTypes.string,
     invalidErrorMessage: PropTypes.string,
     value: PropTypes.string,
-    isValid: PropTypes.func
+    isValid: PropTypes.bool
```
Comment (hwigmore) [inline react/input/TextFieldWithDropdown.jsx]: updating this to reflect the  `TextField` component it points to

File: react/input/TextArea.jsx (hunk 1 above)
Comment (Jinsoo94) [reply react/input/TextArea.jsx]: hmmm for some reason I cant see the height change in the deploy preview. It looks like the master.

**Comment by Jinsoo94** [comment]
In the preview, the textfield with dropdown alignment is off like this!
screen shot 2018-12-07 at 12 52 45 pm

File: react/input/TextArea.jsx (hunk 1 above)
Comment (hwigmore) [reply react/input/TextArea.jsx]: are you looking at the react component or the plain html example? i did the same thing but then i scrolled down and found it!

**Comment by hwigmore** [comment]
good catch! since that's on master and out of scope i'll add a bug ticket for it!

File: react/input/TextArea.jsx (hunk 1 above)
Comment (Jinsoo94) [reply react/input/TextArea.jsx]: ohhh you are right. React component looks correct. Just HTML example and edge case example seem not to change height.

**Comment by hwigmore** [comment]
@Jinsoo94 any other thoughts?

**Comment by Jinsoo94** [comment]
Oh oops forgot give a thumbsup. All good! 👍

**Comment by hwigmore** [comment]
@Jinsoo94 just an FYI that my most recent pr fixed this

**Comment by Jinsoo94** [comment]
@hwigmore ok just checked it 👍

**Pull request merged**